### PR TITLE
Structured entry requirements

### PR DESCRIPTION
**Pull request by frankieroberto**
This changes the "Qualifications" entry requirements from being a simple text box, to something more structured.

The aim is to only capture the differences between courses, in a structured way, rather than requiring each course to state the standard entry requirements in their own words.

This should make it easier to list courses, and will also enable us to be more consistent in how we present this information to candidates. We would also be able to use the structured information to warn candidates whose qualifications do not meet the requirements. We could also introduce filters enabling candidates to more easily find courses for which they meet the entry criteria.

#### Course summary (first tab)
qa publish-teacher-training-courses service gov uk_organisations_1OZ_2021_courses_28TG

#### Free text field for qualifications needed
publish-courses-prototype herokuapp com_course_2CH_2YTX_requirements

#### Options for GCSEs (UCAS only)
qa publish-teacher-training-courses service gov uk_organisations_1OZ_2021_courses_28TG_entry-requirements

### Personal qualities and requirements
localhost_3000_course_2CH_2YTL_requirements

### Degree - do you require a minimum classification?
localhost_3000_course_2CH_2YTL_degree

### Degree - What is the minimum classification required? (if Yes)
localhost_3000_course_2CH_2YTL_degree-level

### Degree subject (non-Primary courses)
localhost_3000_course_2CH_2YTL_degree-subject

### GCSE equivalency tests
localhost_3000_course_2CH_2YTL_gcses-pending-or-equivalency-tests

### Summary on course page
localhost_3000_course_2CH_2YTL (3)

### CODE REVIEW

**Comment by frankieroberto** [comment]
[📐 Publish: Prototype managing structured entry requirements](https://trello.com/c/RWx4Ffms/3295-%F0%9F%93%90-publish-prototype-managing-structured-entry-requirements)

File: app/views/course/requirements.html [hunk 1]
```diff
@@ -1,62 +1,24 @@
...
-          <li>what qualifications they need to teach in England</li>
-          <li>where to go to find out about international qualifications</li>
-          <li>why they might need to do a subject knowledge enhancement course</li>
-        </ul>
-
-        {% set qualificationsHtml %}
-          <h3 class="govuk-!-margin-top-0">Qualifications</h3>
-
-          <p>To be a teacher in England you need:</p>
+      <h1 class="govuk-heading-l">
```
Comment (jesseyuen) [inline app/views/course/requirements.html:10]: Is the indentation a bit off here?

**Comment by paulrobertlloyd** [comment]
Looks good. Few things to consider, maybe in separate PRs:

* Should the structured qualifications content appear within the greyed out boxes on Find?
* This all looks good from an editing point of view, curious how this would work during rollover (still trying to figure this all out myself).
* Probably need a separate PR in which we mention somewhere about adding/updating this information, alongside where we talk about other changes providers need to make on rollover.
* Be good to collate a list of questions we want to ask when showing this to providers, in particular around equivalencies. I’m curious how closely we need to continue giving providers the same options currently provided by UCAS.

**Comment by frankieroberto** [comment]
@paulrobertlloyd yeah, I started thinking a bit about rollover, but wanted to look at how he handle this across other sections. It looked like there were a few patterns already in place?

There’s also the new course flow, which I’m assuming would step through each of the questions in turn. Doesn’t fully work in the prototype yet.

I kept the GCSE equivalencies question as compatible with UCAS for now (albeit reworded), partly for ease of rollover, and partly because it seems there might be some differences across subjects (eg equivalency tests offered in science but not maths). But it’s be be good to probe this in research - and also understand the policy a bit more.

**Review by jesseyuen — APPROVED** [review]

**Comment by jesseyuen** [comment]
There's quite a lot of new data to collect for the next cycle. We
should chat more about our options for sign posting this.

**Pull request merged**